### PR TITLE
DT-618: Add in spring security

### PR DESCRIPTION
**Pull request by petergphillips**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/tokenverification/service/TokenService.kt [hunk 1]
```diff
@@ -40,7 +41,7 @@ class TokenService(private val tokenRepository: TokenRepository) {
   fun revokeTokens(authJwtId: String) {
     log.info("Removing all tokens for authJwtId of {}", authJwtId)
     val tokens = tokenRepository.findByAuthJwtId(authJwtId)
-    log.debug("Deleting ${tokens.size} for auth token")
+    log.debug("Deleting {} for auth token", tokens.size)
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/tokenverification/service/TokenService.kt]: comment from previous PR

File: src/main/resources/application.yml [hunk 2]
```diff
@@ -11,8 +7,6 @@ spring:
     serialization:
       WRITE_DATES_AS_TIMESTAMPS: false
 
-  thymeleaf.render-hidden-markers-before-checkboxes: true
```
Comment (petergphillips) [inline src/main/resources/application.yml]: no thymeleaf in this project

File: src/main/resources/application.yml [hunk 3]
```diff
@@ -22,12 +16,6 @@ server:
     remote-ip-header: x-forwarded-for
     /protocol_header: x-forwarded-proto
 
-logging:
```
Comment (petergphillips) [inline src/main/resources/application.yml]: we've got logback configuration, so this was just confusing as my logback changes didn't do anything

File: src/test/kotlin/uk/gov/justice/digital/hmpps/tokenverification/resource/TokenResourceRedisTest.kt [hunk 4]
```diff
@@ -2,17 +2,12 @@ package uk.gov.justice.digital.hmpps.tokenverification.resource
...
 
-@Suppress("ClassName")
 @ExtendWith(RedisExtension::class)
 class TokenResourceRedisTest : IntegrationTest() {
-  private val jwtHelper = JwtAuthHelper()
-
   @Test
   fun `workflow for token`() {
-    val jwt = jwtHelper.createJwt(JwtParameters(subject = "bob"))
+    val jwt = jwtHelper.createJwt(subject = "bob")
```
Comment (petergphillips) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/tokenverification/resource/TokenResourceRedisTest.kt]: didn't need a data class for parameters as can do defaults etc on the method instead

File: src/test/kotlin/uk/gov/justice/digital/hmpps/tokenverification/resource/TokenResourceTest.kt [hunk 5]
```diff
@@ -1,114 +1,148 @@
...
 import java.util.*
 
-@Suppress("ClassName")
 class TokenResourceTest : IntegrationTest() {
   @MockBean
   private lateinit var tokenRepository: TokenRepository
 
-  private val jwtHelper = JwtAuthHelper()
-
-  @Nested
```
Comment (petergphillips) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/tokenverification/resource/TokenResourceTest.kt]: nested classes don't work well with mock beans as verifying goes all wrong

File: src/test/kotlin/uk/gov/justice/digital/hmpps/tokenverification/service/JwtAuthHelper.kt [hunk 6]
```diff
@@ -1,12 +1,14 @@
-@file:Suppress("DEPRECATION")
```
Comment (petergphillips) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/tokenverification/service/JwtAuthHelper.kt]: from previous PR - unnecessary

File: src/test/kotlin/uk/gov/justice/digital/hmpps/tokenverification/service/JwtAuthHelper.kt [hunk 7]
```diff
@@ -22,24 +24,25 @@ class JwtAuthHelper {
     keyPair = gen.generateKeyPair()
   }
 
-  fun createJwt(parameters: JwtParameters): String {
+  @Bean
+  fun jwtDecoder(): JwtDecoder = NimbusJwtDecoder.withPublicKey(keyPair.public as RSAPublicKey).build()
```
Comment (petergphillips) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/tokenverification/service/JwtAuthHelper.kt]: new approach - use dynamically generated public key for decoder bean and inject into integration tests

File: src/test/kotlin/uk/gov/justice/digital/hmpps/tokenverification/service/JwtAuthHelper.kt [hunk 8]
```diff
@@ -22,24 +24,25 @@ class JwtAuthHelper {
...
+        .setId(jwtId)
+        .setSubject(subject)
         .addClaims(claims)
-        .setExpiration(Date(System.currentTimeMillis() + parameters.expiryTime.toMillis()))
+        .setExpiration(Date(System.currentTimeMillis() + expiryTime.toMillis()))
         .signWith(SignatureAlgorithm.RS256, keyPair.private)
         .compact()
   }
-
-  data class JwtParameters(val subject: String,
```
Comment (petergphillips) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/tokenverification/service/JwtAuthHelper.kt]: unnecessary data class

File: src/test/kotlin/uk/gov/justice/digital/hmpps/tokenverification/service/JwtAuthHelper.kt (hunk 7 above)
Comment (mikehalmamoj) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/tokenverification/service/JwtAuthHelper.kt]: Nice.

**Pull request merged**